.github/workflows/kserve_m2m_test.yaml

```diff
@@ -162,58 +155,240 @@ jobs:
           echo "Success: No istio-init containers found, CNI is working correctly"
         fi
 
-    - name: Run kserve tests with m2m token from SA kubeflow-user-example-com/default-editor
+    # Test 1: External access with valid token from correct namespace, Must Pass
+    - name: Test external access with valid token
       run: |
         export KSERVE_INGRESS_HOST_PORT=localhost:8080
         export KSERVE_M2M_TOKEN="$(kubectl -n kubeflow-user-example-com create token default-editor)"
-        cd ./apps/kserve/tests && pytest . -vs --log-level info
 
-    - name: Detailed KServe Access Diagnostics
+        RESPONSE=$(curl -s -w "%{http_code}" \
+          -H "Host: isvc-sklearn.kubeflow-user-example-com.example.com" \
+          -H "Authorization: Bearer ${KSERVE_M2M_TOKEN}" \
+          -H "Content-Type: application/json" \
+          "http://${KSERVE_INGRESS_HOST_PORT}/v1/models/isvc-sklearn:predict" \
+          -d '{"instances": [[6.8, 2.8, 4.8, 1.4]]}')
+
+        HTTP_CODE=${RESPONSE: -3}
+        RESPONSE_BODY=${RESPONSE:0:${#RESPONSE}-3}
+
+        if [[ "$HTTP_CODE" == "200" && "$RESPONSE_BODY" == *"predictions"* ]]; then
+          echo "Test passed: Valid token access successful with code $HTTP_CODE"
+        else
+          echo "Test failed: Valid token access returned $HTTP_CODE"
+          echo "Response: $RESPONSE_BODY"
+          exit 1
+        fi
```

Comment:
Please no if else and many echoes, keep it simple. Just fail the whole pipeline if the test fails. That should work with curl --fail --show-errors